### ISSUE TITLE
Text highlight incosistent with native android

### ISSUE THREAD

**Issue by bibaswan-bhawal** [flutter/flutter#124485]
## Steps to Reproduce
highlight text in a textfield

**Expected results:** 
![Screenshot_2023-04-09-06-13-09-54_1c337646f29875672b5a61192b9010f9~2](https://user-images.githubusercontent.com/67685395/230801195-1b5b796d-356c-4254-9ef0-6b72f9d27fe1.jpg)

**Actual results:** 
![Screenshot_2023-04-09-06-13-25-40_8626ae07c94127782067f6806ecec2c5~2](https://user-images.githubusercontent.com/67685395/230801198-6192e8ee-a88b-4c62-9ff6-3c10197767f2.jpg)

The magnification in Flutter when highlighting text in a Textfield is not consistent with the native android implementation

**Comment by huycozy** [comment]
Hi @bibaswan-bhawal 
Do you mean the shape of magnification or the selection by word as mentioned at https://github.com/flutter/flutter/issues/104603? 
Please describe the issue in detail so that we may verify the issue properly. Thanks!

**Comment by bibaswan-bhawal** [comment]
Hi sorry, I typed that up on the bus 😂. Yes, the shape of the magnification is different on Android and ios but it seems Flutter uses the same one regardless. Also, here are some more minute details that I noticed:

- The handlebars when selecting a block of text or under the cursor when scrubbing fades out when within the area of magnification. This is a behaviour both on Android and on ios natively. However, in Flutter they are still visible. 

- when selecting text on ios if shrink the selection smaller than one character it transitions into a cursor natively. On Flutter, it does not let you select less than one character.

- On ios, the magnification will slide up and down based on if your finger is above the text. In Flutter, the magnifier will slide up the first time and then disappear without any animation when my finger moves away from the text but will never show again when it is over the text unless I lift my finger and press and hold again.

- When dragging the cursor around in the text field there it can slightly move up and down as well as the entire width of the text field rather than the length of the text on ios. This is not true in Flutter.

I know these are minor details but they make for a much smoother experience and stay consistent with the native platform implementations.

I will try to add examples of each later today.
